packages/react-reconciler/src/__tests__/useMemoCache-test.js

```diff
@@ -276,8 +276,13 @@ describe('useMemoCache()', () => {
       // this triggers a throw.
       setN(1);
     });
-    expect(root).toMatchRenderedOutput('Count 0 (n=1)');
-    expect(Text).toBeCalledTimes(2);
+    if (gate(flags => flags.enableUnifiedSyncLane)) {
+      expect(root).toMatchRenderedOutput(<h1>Something went wrong.</h1>);
+      expect(Text).toBeCalledTimes(1);
+    } else {
+      expect(root).toMatchRenderedOutput('Count 0 (n=1)');
+      expect(Text).toBeCalledTimes(2);
+    }
```

Comment:
These assertions implicitly tested the error recovery. In concurrent updates we'd retry and this component only throws once:
```tsx
if (shouldFail) {
  shouldFail = false;
  throw new Error('failed');
}
```